homeassistant/components/go2rtc/server.py

```diff
@@ -74,8 +97,54 @@ async def start(self) -> None:
         except TimeoutError as err:
             msg = "Go2rtc server didn't start correctly"
             _LOGGER.exception(msg)
-            await self.stop()
-            raise HomeAssistantError("Go2rtc server didn't start correctly") from err
+            await self._stop()
+            raise Go2RTCServerStartError from err
+
+    async def _watchdog(self) -> None:
+        """Keep respawning go2rtc servers.
+
+        A new go2rtc server is spawned if the process terminates or the API
+        stops responding.
+        """
+        while not self._stop_requested:
+            try:
+                async with asyncio.TaskGroup() as tg:
+                    tg.create_task(self._monitor_process())
+                    tg.create_task(self._monitor_api())
+            except ExceptionGroup as grp:
+                if not grp.subgroup(Go2RTCWatchdogError):
+                    _LOGGER.exception("Watchdog got unexpected exception")
+                if self._stop_requested:
+                    continue  # type: ignore[unreachable]
+                await asyncio.sleep(_RESPAWN_COOLDOWN)
+                try:
+                    await self._stop()
+                    _LOGGER.debug("Spawning new go2rtc server")
+                    with suppress(Go2RTCServerStartError):
+                        await self._start()
+                except Exception:
+                    _LOGGER.exception("Unexpected error when restarting go2rtc server")
+
+    async def _monitor_process(self) -> None:
+        """Raise if the go2rtc process terminates."""
+        _LOGGER.debug("Monitoring go2rtc server process")
+        if self._process:
+            await self._process.wait()
+        _LOGGER.debug("go2rtc server terminated")
+        raise Go2RTCWatchdogError("Process ended")
+
+    async def _monitor_api(self) -> None:
+        """Raise if the go2rtc process terminates."""
+        client = Go2RtcRestClient(async_get_clientsession(self._hass), DEFAULT_URL)
+
+        _LOGGER.debug("Monitoring go2rtc API")
+        try:
+            while True:
+                await client.streams.list()
```

Comment:
Is this the simplest request we can do?

Reply:
I think we should replace it with validating the version is supported

Reply:
I'd prefer to do this in a follow-up though